File: Codeforces/Codeforces Round #789/B_2_Tokitsukaze_and_Good_01_-String_hard_version.py
# Codeforces 1678B1  Tokitsukaze and Good 01-String (hard version)
# https://codeforces.com/contest/1678/problem/B2

import sys
input = sys.stdin.readline
from array import array


def solve():
    n = int(input())
    s = list(input().rstrip())
    count = 0
    minSegs = 0
    LAST = -1
    for i in range(0, len(s), 2):
        if s[i]!=s[i+1]:
            count += 1
        else:
            minSegs +=  s[i] != LAST
            LAST = s[i]
            
    return count, max(minSegs, 1)

for _ in range(int(input())):

    print(*solve())

File: Codeforces/Codeforces Round #789/B_2_Tokitsukaze_and_Good_01_-String_hard_version.cpp
// Codeforces 1678B1  Tokitsukaze and Good 01-String (hard version)
// https://codeforces.com/contest/1678/problem/B2

#include "bits/stdc++.h"
using namespace std;

void solve() {
    int n; cin >> n;
    string s; cin >> s;
    int minOps = 0;
    int minSegs = 0;
    int last = -1;
    for (int i = 0; i < n; i += 2) {
        if (s[i] != s[i+1]) {
            minOps++;
        } else {
            if (s[i]!=last) {
                minSegs++;
            }
            last = s[i];
        }
    }
    cout << minOps << " " << max(1, minSegs) << endl;
}


int main() {
    int t; cin >> t;
    while (t--){
        solve();
    }
}